src/ray/object_manager/plasma/create_request_queue.cc

```diff
@@ -93,6 +101,12 @@ Status CreateRequestQueue::ProcessRequests() {
     bool spilling_required = false;
     auto status =
         ProcessRequest(/*fallback_allocator=*/false, *request_it, &spilling_required);
+
+    if (MayHandleOutOfDisk(*request_it)) {
```

Comment:
Personal NIT, but I feel like it might be easier to understand if we don't have `MayHandleOutOfDisk `?

```cpp
  if (request_it->error == PlasmaError::OutOfMemory && fs_monitor_.OverCapacity()) {
    request_it->error = PlasmaError::OutOfDisk;
    FinishRequest(request_it)
  }

```